### PR TITLE
[no-release-notes] Add test for nil bound value panic

### PR DESCRIPTION
**Pull request by max-hoffman**
re: https://github.com/dolthub/go-mysql-server/pull/2581

Stats test with a nil bound value. It took a bit to repro because we always test with reloaded stats, which zero bound value nulls. Add a suite for stats tests without reloading.

### CODE REVIEW

File: go/libraries/doltcore/sqle/enginetest/stats_queries.go [hunk 1]
```diff
@@ -331,6 +331,23 @@ var DoltStatsIOTests = []queries.ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "nulls panic bug",
```
Comment (nicktobey) [inline go/libraries/doltcore/sqle/enginetest/stats_queries.go:335]: Can you add more context about what is being tested here? It could be a more detailed name, a comment, or a link to an issue.

**Review by nicktobey — APPROVED** [review]

**Comment by coffeegoddd** [comment]
@max-hoffman __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>be83aa7</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>be83aa7</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>
